src/components/ConcurrencyLimitsV2UpdateModal.vue

```diff
@@ -63,34 +60,23 @@
     isGreaterThanZeroOrNull,
   ])
 
-  const active = ref(true)
-
-  const activeSlots = ref(props.concurrencyLimit.activeSlots)
-
-
-  const decay = ref(props.concurrencyLimit.slotDecayPerSecond)
-
-
-  const internalShowModal = computed({
-    get() {
-      return props.showModal
-    },
-    set(value: boolean) {
-      emit('update:showModal', value)
-    },
-  })
+  const active = ref(props.concurrencyLimit.active ?? true)
+  const activeSlots = ref(props.concurrencyLimit.activeSlots ?? 0)
+  const decay = ref(props.concurrencyLimit.slotDecayPerSecond ?? 0)
 
   const api = useWorkspaceApi()
   const concurrencyLimitSubscription = useSubscription(api.concurrencyV2Limits.getConcurrencyV2Limits)
 
   const reset = (): void => {
     name.value = props.concurrencyLimit.name
     limit.value = props.concurrencyLimit.limit
-    decay.value = props.concurrencyLimit.slotDecayPerSecond ?? 0
     active.value = props.concurrencyLimit.active ?? true
     activeSlots.value = props.concurrencyLimit.activeSlots ?? 0
+    decay.value = props.concurrencyLimit.slotDecayPerSecond ?? 0
   }
 
+  watch(() => props.concurrencyLimit, reset)
```

Comment:
reset the form fields when the prop is updated - i.e. update the limit from the table(flip active toggle) and then open the modal